lib/cartopy/io/shapereader.py

```diff
@@ -429,9 +427,9 @@ def zip_file_contents(self, format_dict):
 
         """
         for ext in ['.shp', '.dbf', '.shx']:
-            yield (os.path.join('GSHHS_shp', '{scale}',
-                                'GSHHS_{scale}_L{level}{extension}'
-                                ).format(extension=ext, **format_dict))
+            p = Path('GSHHS_shp') / '{scale}'
+            p /= 'GSHHS_{scale}_L{level}{extension}'
```

Comment:
Note that this could be written together:
```suggestion
            p = Path('GSHHS_shp', '{scale}',
                     'GSHHS_{scale}_L{level}{extension}')
```

Reply:
Thanks, updated!